src/main/java/bio/terra/service/filedata/DrsService.java

```diff
@@ -557,36 +561,82 @@ private DRSAccessURL signAzureUrl(
   }
 
   private DRSAccessURL signGoogleUrl(
-      SnapshotCacheResult cachedSnapshot, String gsPath, AuthenticatedUserRequest authUser) {
-    Storage storage;
+      SnapshotCacheResult cachedSnapshot,
+      String gsPath,
+      AuthenticatedUserRequest authUser,
+      String userProject) {
+    BlobId locator = GcsUriUtils.parseBlobUri(gsPath);
+
+    BlobInfo blobInfo = BlobInfo.newBuilder(locator).build();
+
+    String signingProject;
+    String signingUser;
```

Comment:
Normally I don't think it's worth using `final` on local variables since we rarely reassign them, but I make the exception when declaring but not initializing a variable. In that case I would use `final` to ensure that they're only initialized once, and relieve the reader from having to check to see if the initialized value is later overwritten.
```suggestion
    final String signingProject;
    final String signingUser;
```

Reply:
Yeah, I was debating.  They're effectively final (and the function below will squawk without the initialization) but I think it's a good call just for readability to have the final in there.  I can update